calculator_test.go

```diff
@@ -5,20 +5,82 @@ import (
 	"testing"
 )
 
-func TestAdd(t *testing.T) {
-	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
-	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+func TestCalculator(t *testing.T) {
+
+	var tests = []struct {
+		function func(float64, float64) float64
+		want     float64
+		input    []float64
+	}{
+		{function: calculator.Add, input: []float64{2, 2}, want: 4},
+		{function: calculator.Subtract, input: []float64{2, 4}, want: 2},
+		{function: calculator.Multiply, input: []float64{2, 4}, want: 8},
+	}
+	for _, tt := range tests {
+		got := tt.function(tt.input[0], tt.input[1])
+		if tt.want != got {
+			t.Errorf("want %f, got %f", tt.want, got)
+		}
+
+	}
+}
+
+func TestDivide(t *testing.T) {
+	var tests = []struct {
+		a             float64
+		b             float64
+		want          float64
+		errorExpected bool
+	}{
+		{
+			a:             4,
+			b:             0,
+			want:          999,
```

Comment:
Good! This seems to be a little-known trick; at least, I've never seen anyone else do it. But it's great for catching bugs in the test. If you see a `999` in the output, something's gone wrong!